### PR TITLE
mon/OSDMonitor: increase last_epoch_clean's lower bound if possible

### PR DESCRIPTION
**Pull request by tchaikov**
Signed-off-by: Kefu Chai <kchai@redhat.com>

### CODE REVIEW

File: src/mon/OSDMonitor.cc [hunk 1]
```diff
@@ -80,13 +80,21 @@ void LastEpochClean::Lec::report(ps_t ps, epoch_t last_epoch_clean)
...
+  const auto old_lec = epoch_by_pg[ps];
+  if (old_lec >= last_epoch_clean) {
     // stale lec
     return;
   }
   epoch_by_pg[ps] = last_epoch_clean;
   if (last_epoch_clean < floor) {
     floor = last_epoch_clean;
+  } else if (last_epoch_clean > floor) {
+    if (old_lec == floor) {
```
Comment (gregsfortytwo) [inline src/mon/OSDMonitor.cc]: Probably clearer to combine these two if statements into one? As is it kinda looks like there should be a change even if old_lec != floor.

Either way, LGTM.

Comment (tchaikov) [inline src/mon/OSDMonitor.cc]: @gregsfortytwo actually i thought about this. but it seems that it maintains better symmetry (compares `last_epoch_clean` with `floor`) this way.

Comment (gregsfortytwo) [inline src/mon/OSDMonitor.cc]: ```else if (last_epoch_clean > floor && old_lec == floor)```
Is what I meant.
But, either way.

Comment (tchaikov) [inline src/mon/OSDMonitor.cc]: yes, i did get you.

**Pull request merged**